src/axom/sina/core/CurveSet.cpp

```diff
@@ -35,63 +37,118 @@ constexpr auto DEPENDENT_KEY = "dependent";
 /**
  * Add a curve to the given curve map.
  *
+ * Helper function, users use addIndependentCurve/addDependentCurve.
+ *
  * @param curve the curve to add
  * @param curves the CurveMap to which to add the curve
+ * @param nameList the vector of curve names to add the curve's name to.
+                   Used for tracking insertion order for codes.
  */
-void addCurve(Curve &&curve, CurveSet::CurveMap &curves)
+void addCurve(Curve &&curve, CurveSet::CurveMap &curves, std::vector<std::string> &nameList)
 {
   auto &curveName = curve.getName();
   auto existing = curves.find(curveName);
   if(existing == curves.end())
   {
     curves.insert(std::make_pair(curveName, curve));
+    nameList.emplace_back(curveName);
```

Comment:
Should this be `push_back` since presumably, `curveName` still belongs to `curve` ?